src/__test__/TradesDetails.test.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import {
+  render, screen, waitFor, fireEvent,
+} from '@testing-library/react';
+import { MemoryRouter, Routes, Route } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import TradesDetails from '../components/DetailsTrades';
+
+const mockStore = configureStore([thunk]);
+
+describe('TradesDetails', () => {
+  let store;
+  const tradeData = {
+    id: 1,
+    name: 'Trade Name',
+    description: 'Trade Description',
+    location: 'Trade Location',
+    price: 100,
+    duration: '1 hour',
+    trade_type: 'Type of Trade',
+    image: 'image-url',
+  };
+
+  beforeEach(() => {
+    store = mockStore({
+      tradeDetails: {
+        trade: tradeData,
+      },
+    });
+  });
+
+  it('renders trade details when trade data is available', async () => {
+    render(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={['/trade/1']}>
+          <Routes>
+            <Route path="/trade/:id" element={<TradesDetails />} />
+          </Routes>
+        </MemoryRouter>
+      </Provider>,
+    );
+
+    await waitFor(() => screen.getByText('Trade Name'));
+
+    expect(screen.getByText('Trade Name')).toBeInTheDocument();
+    expect(screen.getByText('Trade Description')).toBeInTheDocument();
+    expect(screen.getByText('Trade Location')).toBeInTheDocument();
+    expect(screen.getByText(/\$100\b/)).toBeInTheDocument();
+    expect(screen.getByText('1 hour hours')).toBeInTheDocument();
+    expect(screen.getByText('Type of Trade')).toBeInTheDocument();
+
+    expect(screen.getByText('Click to Reserve')).toBeInTheDocument();
+    const reserveButton = screen.getByText('Click to Reserve');
+    fireEvent.click(reserveButton); // Click the button
+  });
+
+  it('displays loading message when trade data is not available', async () => {
+    store = mockStore({
+      tradeDetails: {
+        trade: null,
+      },
+    });
+
+    render(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={['/trade/1']}>
+          <Routes>
+            <Route path="/trade/:id" element={<TradesDetails />} />
```

Comment:
I got an error in this line.

![Screenshot from 2023-10-09 12-31-39](https://github.com/lincoln1883/appointment-capstone-frontend/assets/84746035/de4ca922-2f6c-4024-a356-dd244cd724c9)

Reply:
please install the dependencies first by `npm install` before running the tests, its working in my environment